---
author: Elizabeth Czarniak (CZARNIA_ELIZ@bentley.edu)
---

The method we will use is equivalent to subtracting the two samples and then
performing the signed-rank test.  See how to do a Wilcoxon signed-rank test to
compare the two methods.

We're going to use fake data for illustrative purposes,
but you can replace our fake data with your real data.

In [6]:
import numpy as np
# Replace sample1 and sample2 with your data
sample1 = np.array([156, 133, 90, 176, 119, 120, 40, 52, 167, 80])
sample2 = np.array([45, 36, 78, 54, 12, 25, 39, 48, 52, 70])

We choose a value, $0 \le \alpha \le 1$, as the Type I Error Rate. We'll let $\alpha$ be 0.05.

### Two-tailed test

To test the null hypothesis $H_0: m_D = 0$, we use a two-tailed test:

In [7]:
from scipy import stats
from scipy.stats import wilcoxon
wilcoxon(sample1 - sample2)

WilcoxonResult(statistic=0.0, pvalue=0.001953125)

Our p-value, 0.001953125, is less than $\alpha=0.05$, so we have sufficient evidence
to reject the null hypothesis. The median difference is significantly different
from zero.

### Right-tailed test

To test the null hypothesis $H_0: m_D \le 0$, we use a right-tailed test:

In [10]:
wilcoxon(sample1 - sample2, alternative = 'greater')

WilcoxonResult(statistic=55.0, pvalue=0.0009765625)

Our p-value, 0.0009765625, is less than $\alpha=0.05$, so we have sufficient evidence
to reject the null hypothesis. The median difference is significantly greater
than zero.

### Left-tailed test

To test the null hypothesis $H_0: m_D \ge 0$, we use a left-tailed test:

In [9]:
wilcoxon(sample1 - sample2, alternative = 'less')

WilcoxonResult(statistic=55.0, pvalue=1.0)

Our p-value, 1.0, is greater than $\alpha$, so we do not have sufficient
evidence to reject the null hypothesis. We should continue to assume that the
mean difference may be less than (or equal to) zero.